Fix invalid permission key to solve error on update files via API

hissy opened the pull request:
When you try to update files via API, you'll get the following error:

```
Exception: Unable to get permission key for edit_file in file /path/to/concrete/src/Permission/Response/Response.php on line 107
```

This PR will fix it.

hissy committed:
Fix invalid permission key to solve error on update files via API

mlocati commented:
There's another `$checker->canEditFile()` call [later on in the file](https://github.com/concretecms/concretecms/blob/12f45e168e8b8d8bede12cee03ad0103ea46914a/concrete/src/Api/Controller/Files.php#L377): should it be fixed too?

hissy commented:
@mlocati 

I think so, but I don't know the proper permission key to check. "Add File" permission for the target folder?

mlocati commented:
If we move a file from folder A to another B one is like deletiing the file from A and adding it to B.

So, what about replacing

```php
(new Checker($file))->canEditFile()
```

with

```php
(new Checker($file))->canEditFileProperties()
&&
(new Checker($sourceFolder))->canDeleteFileFolderFiles()
&&
(new Checker($destinationFolder))->canAddFile()
```

hissy committed:
Check proper permissions. edit_file is an invalid permission key.

Changed to check the same permission keys as the file manager. See \Concrete\Controller\Dialog\File\Folder::submit

hissy commented:
I tried to move a file by API but got an error message saying, "Unable to get permission key for edit_file."
Changed to check the same permission keys as the file manager. See https://github.com/concretecms/concretecms/blob/9.3.x/concrete/controllers/dialog/file/folder.php
